modular_skyrat/modules/modular_implants/code/soulcatcher/handheld_soulcatcher.dm

```diff
@@ -111,4 +111,52 @@
 
 	return TRUE
 
+/obj/item/handheld_soulcatcher/attack_secondary(mob/living/carbon/human/target_mob, mob/living/user, params)
+	if(!istype(target_mob))
+		return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+
+	var/obj/item/organ/internal/brain/target_brain = target_mob.get_organ_slot(ORGAN_SLOT_BRAIN)
+	if(!istype(target_brain))
+		to_chat(user, span_warning("[target_mob] lacks a brain!"))
+		return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+
+	if(!HAS_TRAIT(target_brain, TRAIT_RSD_COMPTAIBLE))
+		to_chat(user, span_warning("[target_mob]'s brain isn't compatible."))
+		return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+
+	if(target_mob.mind || target_mob.ckey || GetComponent(/datum/component/previous_body))
+		to_chat(user, span_warning("[target_mob] is not able to recieve a soul"))
+		return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+
+	var/list/soul_list = list()
+	for(var/datum/soulcatcher_room/room as anything in linked_soulcatcher.soulcatcher_rooms)
+		for(var/mob/living/soulcatcher_soul/soul as anything in room.current_souls)
+			if(!soul.round_participant || soul.body_scan_needed)
+				continue
+
+			soul_list += soul
```

Comment:
I have another PR that has a proc that gets a list of all of the souls currently inside of a soulcatcher. If this is something that you believe needs done, it would best be done to that proc.